scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,314 @@
+            Contains the lower and upper bounds of the confidence interval.
+
+        References
+        ----------
+        .. [1] C. J. Clopper and E. S. Pearson, The use of confidence or
+               fiducial limits illustrated in the case of the binomial,
+               Biometrika, Vol. 26, No. 4, pp 404-413 (Dec. 1934).
+        .. [2] E. B. Wilson, Probable inference, the law of succession, and
+               statistical inference, J. Amer. Stat. Assoc., 22, pp 209-212
+               (1927).
+        .. [3] Robert G. Newcombe, Two-sided confidence intervals for the
+               single proportion: comparison of seven methods, Statistics
+               in Medicine, 17, pp 857-872 (1998).
+
+        Examples
+        --------
+        >>> from scipy.stats import binomtest
+        >>> result = binomtest(k=7, n=50, p=0.1)
+        >>> result.proportion_estimate
+        0.14
+        >>> result.proportion_ci
+        ConfidenceInterval(low=0.04246878737883377, high=0.30910696596481874)
+        """
+        if method not in ('exact', 'wilson', 'wilsoncc'):
+            raise ValueError("method must be one of 'exact', 'wilson' or "
+                             "'wilsoncc'.")
+        if not (0 <= confidence_level <= 1):
+            raise ValueError('confidence_level must be in the interval '
+                             '[0, 1].')
+        if method == 'exact':
+            low, high = _binom_exact_conf_int(self.k, self.n,
+                                              confidence_level,
+                                              self.alternative)
+        else:
+            # method is 'wilson' or 'wilsoncc'
+            low, high = _binom_wilson_conf_int(self.k, self.n,
+                                               confidence_level,
+                                               self.alternative,
+                                               correction=method == 'wilsoncc')
+        return ConfidenceInterval(low=low, high=high)
+
+
+def _binom_exact_conf_int(k, n, confidence_level, alternative):
+    """
+    Compute the estimate and confidence interval for the binomial test.
+
+    Returns proportion, prop_low, prop_high
+    """
+    if alternative == 'two-sided':
+        alpha = (1 - confidence_level) / 2
+        if k == 0:
+            plow = 0.0
+        else:
+            plow = brentq(lambda p: binom.sf(k-1, n, p) - alpha, 0, 1)
+        if k == n:
+            phigh = 1.0
+        else:
+            phigh = brentq(lambda p: binom.cdf(k, n, p) - alpha, 0, 1)
+    elif alternative == 'less':
+        alpha = 1 - confidence_level
+        plow = 0.0
+        if k == n:
+            phigh = 1.0
+        else:
+            phigh = brentq(lambda p: binom.cdf(k, n, p) - alpha, 0, 1)
+    elif alternative == 'greater':
+        alpha = 1 - confidence_level
+        if k == 0:
+            plow = 0.0
+        else:
+            plow = brentq(lambda p: binom.sf(k-1, n, p) - alpha, 0, 1)
```

Comment:
To be safe, should we pass `full_output=True` to `brentq` and check that the solver converged, throwing an exception if not?

from the docs in `brentq` it indicates that this solver should always converge so I'm not firm on this one.

Reply:
I modified the code to wrap the calls to `brentq` in a `try/except` statement, so the user will get a more useful error message if `brentq` raises an exception.

Reply:
thanks, marking resolved